components/feature/webcompat/src/main/java/mozilla/components/feature/webcompat/WebCompatFeature.kt

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.webcompat
+
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.webextension.WebExtension
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.log.logger.Logger
+
+class WebCompatFeature(
+    private val engine: Engine
+) : LifecycleAwareFeature {
+
+    override fun start() {
```

Comment:
This feature could be simplified a lot. Since this component has no native UI at all, you could turn this into an `object` instead of a class. And this object would only need an install method.

Where we create the `Engine` we could then also install your feature directly, once. So you wouldn't need that logic here. :)